client/src/utils/components/list/List.tsx

```diff
@@ -0,0 +1,125 @@
+}
+
+export const VisibilityIcon = ({ visibility }: VisibilityIconProps) => {
+  if (!visibility) return null;
+  const icon = {
+    public: <FontAwesomeIcon icon={faGlobe} />,
+    private: <FontAwesomeIcon icon={faLock} />,
+    internal: <FontAwesomeIcon icon={faUserFriends} />
+  };
+  return <span className="visibility-icon text-rk-text" title={toCapitalized(visibility)}>
+    { icon[visibility] || "" }
+  </span>;
+};
+
+/**
+ * ListCard/ListBar returns a card or a bar displaying an item in a List.
+ *
+ * @param type Type of visualization (Card, Bar).
+ * @param url containing a link to the entity details.
+ * @param title title of the entity.
+ * @param slug slug of the entity.
+ * @param description description of the entity.
+ * @param visibility visibility of the entity.
+ * @param tagList tag list of the entity.
+ * @param timeCaption date to put inside the time caption of the entity.
+ * @param labelCaption label to put inside the time caption of the entity, if empty defaults to Updated.
+ * @param mediaContent image of the entity.
+ * @param creators creators of the entity, if more than 3 they will be cropped at 3.
+ * @param itemType type of the entity being rendered, the color of the circle depends on this.
+ * @param handler handler function for main button.
+ */
+function List(
+  { type,
+    url,
+    title,
+    description,
+    tagList,
+    timeCaption,
+    labelCaption,
+    mediaContent,
+    creators,
+    itemType,
+    slug,
+    visibility,
+    handler
+  }: ListProps) {
+
+  const onMainButton = () => {
+    if (handler)
+      handler();
+  };
+
+  const entityIcon = itemType === EntityType.Project ? "/project-icon.svg" : "/dataset-icon.svg";
+  const visibilityIcon = visibility ? <VisibilityIcon visibility={visibility} /> : null;
+  const rightButton = itemType === EntityType.Project ?
+    (<Button key="start-session" color="primary" onClick={onMainButton}>
+      <FontAwesomeIcon className="me-1" icon={faPlay} /> Start session
+    </Button>) :
+    (<Button key="start-session" color="primary" onClick={onMainButton}>
+      <FontAwesomeIcon className="me-1" icon={faPlus} /> Add to project
+    </Button>);
+  return type === ListType.Card ? (
+    <ListCard
+      url={url}
+      title={title}
+      description={description}
+      tagList={tagList}
+      timeCaption={timeCaption}
+      labelCaption={labelCaption}
+      mediaContent={mediaContent}
+      creators={creators}
+      slug={slug}
+      entityIcon={entityIcon}
+      visibilityIcon={visibilityIcon}
+      rightButton={rightButton}
+    />
+  ) : (
+    <ListBar
+      url={url}
+      title={title}
+      description={description}
+      tagList={tagList}
+      timeCaption={timeCaption}
+      labelCaption={labelCaption}
+      mediaContent={mediaContent}
+      creators={creators}
+      slug={slug}
+      entityIcon={entityIcon}
+      visibilityIcon={visibilityIcon}
+      rightButton={rightButton}
+    />
+  );
+}
```

Comment:
With the changes above, this can be written as:
```suggestion
  return type === ListType.Card ? (
    <ListCard entityIcon={entityIcon} visibilityIcon={visibilityIcon} rightButton={rightButton} {...props} />
  ) : (
    <ListBar entityIcon={entityIcon} visibilityIcon={visibilityIcon} rightButton={rightButton} {...props} />
  );
}
```